storage/engine/engine.go

```diff
@@ -188,6 +188,13 @@ func PutProto(engine Engine, key proto.EncodedKey, msg gogoproto.Message) (keyBy
 // key was not found. On success, returns the length in bytes of the
 // key and the value.
 func GetProto(engine Engine, key proto.EncodedKey, msg gogoproto.Message) (ok bool, keyBytes, valBytes int64, err error) {
+	type protoGetter interface {
```

Comment:
I decided to just add `GetProto` to `Engine` and `ValueProto` to `Iterator`. Not as general as the `Unmarshaler` idea, but simpler.